lib/internal/test_runner/yaml_parser.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+const {
+  codes: {
+    ERR_TEST_FAILURE,
+  }
+} = require('internal/errors');
+const AssertionError = require('internal/assert/assertion_error');
+const {
+  ArrayPrototypeJoin,
+  Error,
+  Number,
+  NumberIsNaN,
+  RegExpPrototypeExec,
+  StringPrototypeEndsWith,
+  StringPrototypeRepeat,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+  StringPrototypeSubstring,
+} = primordials;
+
+const kYamlKeyRegex = /^(\s+)?(\w+):(\s)+([>|][-+])?(.*)$/;
+const kStackDelimiter = '    at ';
```

Comment:
In `tap_stream.js` we already have `const kFrameStartRegExp = /^ {4}at /;`. Since `tap_stream` will be less important in the future, maybe that file could import necessary YAML utils like this from here?

Reply:
I want to reduce potential conflicts so Id rather not doing that here